### PR TITLE
Upgrade Preact to 10.5.13

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

**Comment by dvoytenko** [comment]
@samouri Not sure why renovate didn't trigger this upgrade since it looks like we were quite behind. The reason I'm upgrading is due to this: https://github.com/preactjs/preact/issues/2726

**Comment by dvoytenko** [comment]
@samouri I think I have a suspicion why Renovate didn't go through. It looks like our compiler is failing for some reason. @jridgewell could you PTAL at Travis logs? Something odd. Either there's some optimization weirdness or maybe Preact is using some JS features we don't understand?

File: src/preact/index.js [hunk 1]
```diff
@@ -70,7 +70,8 @@ export function createRef() {
  * @template T
  */
 export function createContext(value) {
-  return preact.createContext(value);
+  // TODO(preactjs/preact#2736): Remove once Preact's API is fixed.
+  return preact.createContext(value, 0);
```
Comment (jridgewell) [inline src/preact/index.js:74]:
```suggestion
  return preact.createContext(value, undefined);
```

**Comment by jridgewell** [comment]
Waiting on cl/329403444 to fix a closure issue.

**Comment by samouri** [comment]
I think this can be revived now that the closure issue has been fixed?

**Comment by rsimha** [comment]
> Is there any particular reason (new features etc) that prompted you to upgrade out-of-band with Renovate?

> Not sure why renovate didn't trigger this upgrade since it looks like we were quite behind. 

> I think I have a suspicion why Renovate didn't go through.

Drive-by comment to answer all three questions above: We explicitly do not auto-update all packages in the `dependencies` section of our root `package.json`. This has been discussed a few times before (https://github.com/ampproject/amphtml/pull/12571, https://github.com/ampproject/amphtml/issues/12181, https://github.com/ampproject/amphtml/pull/23008, https://github.com/ampproject/amphtml/issues/24784), and the compromise we arrived at was to auto-update all `@ampproject` dependencies, but not others.

If you see value in receiving auto-update PRs from renovate for other dependencies that you can manually merge, here's the config paragraph that needs to be changed:

https://github.com/ampproject/amphtml/blob/15af5ee8ae972d7715d3c0459861d86be748d84c/.renovaterc.json#L88-L96

Let me know if there's interest in doing this. We can automatically apply a working group label for PRs that upgrade something in `dependencies`.

**Comment by samouri** [comment]
Just rebased + pushed

File: extensions/amp-base-carousel/1.0/test-e2e/test-grouping.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ describes.endtoend(
       await expect(prop(slide6, 'offsetLeft')).to.equal(expectedScrollPosition);
     });
 
-    it('should move backwards by the advance-count', async () => {
+    it.skip('should move backwards by the advance-count', async () => {
```
Comment (jridgewell) [inline extensions/amp-base-carousel/1.0/test-e2e/test-grouping.js:74]: /cc @caroqliu I had to skip this e2e test in order for this upgrade to pass tests. Can you investigate?

**Comment by rsimha** [comment]
> Not sure why renovate didn't trigger this upgrade since it looks like we were quite behind.

@dvoytenko Updates have been enabled as laid out in https://github.com/ampproject/amphtml/pull/30043#issuecomment-821478569. The upgrade PR can be found at #34146.

File: extensions/amp-base-carousel/1.0/test-e2e/test-grouping.js (hunk 2 above)
Comment (caroqliu) [reply extensions/amp-base-carousel/1.0/test-e2e/test-grouping.js:74]: I had assumed when you tagged me that this investigation could occur post-merge, wanted to clarify if it's blocking us from being able to upgrade?

**Review by rsimha — APPROVED** [review]

**Review by jridgewell — APPROVED** [review]

**Pull request merged**